src/Bundle/Controller/ControllerTrait.php

```diff
@@ -246,6 +247,36 @@ protected function render(string $view, array $parameters = [], Response $respon
         return $response;
     }
 
+    /**
+     * Renders a view and sets the appropriate status code when a form is listed in parameters.
+     *
+     * If an invalid form is found in the list of parameters, a 422 status code is returned.
+     */
+    protected function renderForm(string $view, array $parameters = [], Response $response = null): Response
+    {
+        if (null === $response) {
+            $response = new Response();
+        }
+
+        foreach ($parameters as $k => $v) {
+            if ($v instanceof FormView) {
+                throw new \LogicException(sprintf('Passing a FormView to "%s::renderForm()" is not supported, pass directly the form instead for parameter "%s".', get_debug_type($this), $k));
+            }
+
+            if (!$v instanceof FormInterface) {
+                continue;
+            }
+
+            $parameters[$k] = $v->createView();
+
+            if (200 === $response->getStatusCode() && $v->isSubmitted() && !$v->isValid()) {
+                $response->setStatusCode(422);
```

Comment:
Why not using Resonse::UNPROCESSABLE_ENTITY?

does the code you wrote come from Symfony itself?